### PR TITLE
[5.7] Collection whereBetween 

### PR DESCRIPTION
**Pull request by dmitryuk**
This PR adds posibility to make whereBetween filters in Collections.
`$c = new Collection([['v' => 1], ['v' => 2], ['v' => 3], ['v' => '3'], ['v' => 4]]);`
`$c->whereBetween('v', [2, 4])`

### CODE REVIEW

**Comment by morloderex** [comment]
What about numeric arrays? 

e.g. [1,2,3,4,5,6], it would be a little bit weird to pass in `$collection->whereBetween(null, [2,6])` And would this even work?

**Comment by xakzona** [comment]
> What about numeric arrays?
> 
> e.g. [1,2,3,4,5,6], it would be a little bit weird to pass in `$collection->whereBetween(null, [2,6])` And would this even work?

It working usign where conditions and work like it, i.e. as you can not use where condition with numeric arrays

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -617,6 +617,17 @@ public function whereInStrict($key, $values)
         return $this->whereIn($key, $values, true);
     }
 
+    /**
+     * Filter items where the given key between values.
+     * @param $key
+     * @param $values
+     * @return Collection
```
Comment (driesvints) [inline src/Illuminate/Support/Collection.php]: Please use FQN and add proper types and spacing to the params. Add a white line above the params.

**Review by driesvints — DISMISSED** [review]

File: src/Illuminate/Support/Collection.php (hunk 1 above)
Comment (xakzona) [reply src/Illuminate/Support/Collection.php]: Done

**Comment by LasseRafn** [comment]
Honest question; why not just 3 parameters? $key, $from, $to (or whatever name)

**Comment by LasseRafn** [comment]
Nevermind, I just realized this is the Query Builders logic as well. Ignore me!

**Pull request merged**